test/src/common/bridgePowerController_ut.cpp

```diff
@@ -147,6 +147,287 @@ TEST_F(BridgePowerControllerTest, alignment) {
+
+  // Bus on
+  M = 10;
+  rtcGetMicroSeconds_fake.return_val = 1713233400000000;
+  powerController._update();
+  EXPECT_EQ(fake_io_read_func_fake.call_count, 9);
+  EXPECT_EQ(fake_io_write_func_fake.call_count, 7);
+  EXPECT_EQ(fake_io_write_func_fake.arg1_history[6], 1);
+  curtime += kOneMinute;
+  EXPECT_EQ(xTaskGetTickCount(), curtime);
+
+  // Bus off
+  M = 11;
+  rtcGetMicroSeconds_fake.return_val = 1713233460000000;
+  powerController._update();
+  EXPECT_EQ(fake_io_read_func_fake.call_count, 10);
+  EXPECT_EQ(fake_io_write_func_fake.call_count, 8);
+  EXPECT_EQ(fake_io_write_func_fake.arg1_history[7], 0);
+  curtime += 4 * kOneMinute;
+  EXPECT_EQ(xTaskGetTickCount(), curtime);
+
+  // Bus on
+  M = 15;
+  rtcGetMicroSeconds_fake.return_val = 1713233700000000;
+  powerController._update();
+  EXPECT_EQ(fake_io_read_func_fake.call_count, 11);
+  EXPECT_EQ(fake_io_write_func_fake.call_count, 9);
+  EXPECT_EQ(fake_io_write_func_fake.arg1_history[8], 1);
+  curtime += kOneMinute;
+  EXPECT_EQ(xTaskGetTickCount(), curtime);
+
+  // Bus off, sleep until end of sample duration
+  M = 16;
+  rtcGetMicroSeconds_fake.return_val = 1713233760000000;
+  powerController._update();
+  EXPECT_EQ(fake_io_read_func_fake.call_count, 12);
+  EXPECT_EQ(fake_io_write_func_fake.call_count, 10);
+  EXPECT_EQ(fake_io_write_func_fake.arg1_history[9], 0);
+  curtime += 3 * kOneMinute;
+  EXPECT_EQ(xTaskGetTickCount(), curtime);
+
+  // Stay off and align next sample
+  M = 19;
+  rtcGetMicroSeconds_fake.return_val = 1713233940000000;
+  powerController._update();
+  EXPECT_EQ(fake_io_read_func_fake.call_count, 13);
+  EXPECT_EQ(fake_io_write_func_fake.call_count, 10);
+  curtime += kOneMinute;
+  EXPECT_EQ(xTaskGetTickCount(), curtime);
+
+  // Bus on
+  M = 20;
+  rtcGetMicroSeconds_fake.return_val = 1713234000000000;
+  powerController._update();
+  EXPECT_EQ(fake_io_read_func_fake.call_count, 14);
+  EXPECT_EQ(fake_io_write_func_fake.call_count, 11);
+  EXPECT_EQ(fake_io_write_func_fake.arg1_history[10], 1);
+  curtime += kOneMinute;
+  EXPECT_EQ(xTaskGetTickCount(), curtime);
+
+  // Bus off
+  M = 21;
+  rtcGetMicroSeconds_fake.return_val = 1713234060000000;
+  powerController._update();
+  EXPECT_EQ(fake_io_read_func_fake.call_count, 15);
+  EXPECT_EQ(fake_io_write_func_fake.call_count, 12);
+  EXPECT_EQ(fake_io_write_func_fake.arg1_history[11], 0);
+  curtime += 4 * kOneMinute;
+  EXPECT_EQ(xTaskGetTickCount(), curtime);
+}
+
+// Subsample duration end lines up with sample duration end at 7 minutes.
+// Subsample interval is 3 minutes, which is not 5-minute aligned.
+TEST_F(BridgePowerControllerTest, subsampling2) {
+  const uint32_t kTenMinutes = 600000;
+  const uint32_t kSevenMinutes = 420000;
+  const uint32_t kThreeMinutes = 180000;
+  const uint32_t kOneMinute = 60000;
+  BridgePowerController powerController(FAKE_VBUS_EN, kTenMinutes, kSevenMinutes, kThreeMinutes,
+                                        kOneMinute, true, true);
```

Comment:
I added this test case to ensure I didn't break the existing behavior, and to prove to myself that subsampling is unaffected by UTC alignment.